rfcs/20220221-3d_model_support.md

```diff
@@ -0,0 +1,181 @@
+* The API, while it does have a migration tool, does not have a standardized migration application process. We will need to run the migration manually when the time comes.
+
+## Some other considerations
+
+* It would be ideal to have this under a feature flag in the API. We could perform deployments and testing on our staging setup with the flag enabled, but still make other deployments to production with the flag disabled. We should be able to run the API for images & audio successfully with the feature flag disabled AND without having run the migrations.
+* As we work through this on both the catalog and the API, we should begin compiling information for a document describing the process for adding a **new media type**. This will be useful for future media types we are sure to add!
+
+## Prior work
+
+We may be able to use previous work regarding the addition of audio to the catalog/API as a guide. These updates required a fair amount of generalization, which we may not need to do a second time around.
+
+WordPress/openverse#19
+
+## Technical implementation
+
+### 1. Add 3D models to the catalog
+
+- [ ] Define necessary fields & schema for `model_3d` table & associated data
+- [ ] Add DDL to local Postgres image: https://github.com/WordPress/openverse-catalog/tree/main/docker/local_postgres
+- [ ] Create a popularity calculation view for `model_3d`
+- [ ] Add the database columns to `db_columns.py`: https://github.com/WordPress/openverse-catalog/blob/main/openverse_catalog/dags/common/storage/db_columns.py
+- [ ] Add the TSV columns to `tsv_columns.py`: https://github.com/WordPress/openverse-catalog/blob/main/openverse_catalog/dags/common/storage/tsv_columns.py
+- [ ] Create a `Model3DStore` subclass of [`MediaStore`](https://github.com/WordPress/openverse-catalog/blob/9538f38401e5428b32dc14c5d5ba7ef50af87a96/openverse_catalog/dags/common/storage/media.py#L39): 
+- [ ] Update the [new provider API script template](https://github.com/WordPress/openverse-catalog/tree/main/openverse_catalog/templates) with the `model_3d` provider
+- [ ] Create one/several provider API scripts & associated DAGs
+
+### 2. Add 3D models to the API
+_Note: Borrowed heavily from WordPress/openverse#19_
+- [ ] Django models for `model_3d`
+- [ ] Add DRF serializers for `model_3d`
+- [ ] Add DRF endpoints for `model_3d`
+- [ ] Make the thumbnail server work with model thumbnails
+
+### 3. Add 3D models to the data refresh
+- [ ] Add Elasticsearch `model_3d` models
+- [ ] Ensure the data refresh works for 3D models
+- [ ] Add 3D model sample data into the sample data CSV for testing
+
+
+## Open Questions
+
+* What data is important to store in the catalog? Some of this is encapsulated in the proposed schema, but there may be other important attributes beyond those listed.
+* Will it be important to differentiate the "type" of 3D model that a particular record refers to? (E.g. 3D printer model, game asset, artwork, etc.)
```

Comment:
I think yes, if we can get this metadata it'd be a useful filter :slightly_smiling_face: